src/routes/api/assets/patchById.js

```diff
@@ -1,16 +1,85 @@
 const logger = require("../../../logger");
-const { createSuccessResponse } = require("../../../response");
+const prisma = require("../../../model/data/prismaClient");
+const {
+  createSuccessResponse,
+  createErrorResponse,
+} = require("../../../response");
 
 /**
- * Update an asset by it's id
+ * Update an asset by its id
  */
-
-// eslint-disable-next-line no-unused-vars
 module.exports = async (req, res, next) => {
-  logger.debug(
-    { user: req.user, id: req.params.id },
-    `received request: PATCH /v1/assets/:assetId`,
-  );
+  try {
+    const assetId = Number(req.params.assetId);
+
+    // Add debug logging
+    logger.debug(
+      {
+        params: req.params,
+        assetId: assetId,
+        isNaN: isNaN(assetId),
+        user: req.user,
+        body: req.body,
+      },
+      "PATCH request details",
+    );
+
+    // Validate assetId is a number
+    if (isNaN(assetId)) {
+      return res.status(400).json(createErrorResponse(400, "invalid asset id"));
+    }
+
+    logger.debug(
+      { user: req.user, assetId, body: req.body },
+      "received request: PATCH /v1/assets/:assetId",
+    );
+
```

Comment:
issue: We need to validate the response body before we do anything. We can use Zod to do this. See src/model/asset.js and POST /v1/assets.